site/content/en/docs/tasks/run_python_jobs.md

```diff
@@ -115,117 +61,15 @@ python install-kueue-queues.py
 
-```bash
-kubectl apply -f ../single-clusterqueue-setup.yaml 
-```
-```console
-resourceflavor.kueue.x-k8s.io/default-flavor created
-clusterqueue.kueue.x-k8s.io/cluster-queue created
-localqueue.kueue.x-k8s.io/user-queue created
-```
-
-And now launch a job. This simple example mimics the [sample-job.yaml](https://github.com/kubernetes-sigs/kueue/blob/main/examples/sample-job.yaml) and demonstrates submitting a job and then listing it in the Kueue queue.
-Write this to `sample-job.py`:
-
 ```python
-#!/usr/bin/env python3
-
-import argparse
-from kubernetes import config, client
-
-# create_job.py
-# This example will demonstrate full steps to submit a Job.
-
-# Make sure your cluster is running!
-config.load_kube_config()
-crd_api = client.CustomObjectsApi()
-api_client = crd_api.api_client
-
-
-def get_parser():
-    parser = argparse.ArgumentParser(
-        description="Submit Kueue Job Example",
-        formatter_class=argparse.RawTextHelpFormatter,
-    )
-    parser.add_argument(
-        "--job-name",
-        help="generateName field to set for job",
-        default="sample-job-",
-    )
-    parser.add_argument(
-        "--image",
-        help="container image to use",
-        default="gcr.io/k8s-staging-perf-tests/sleep:v0.0.3",
-    )
-    parser.add_argument(
-        "--args",
-        nargs="+",
-        help="args for container",
-        default=["30s"],
-    )
-    return parser
-
-
-def generate_job_crd(job_name, image, args):
-    """
-    Generate an equivalent job CRD to sample-job.yaml
-    """
-    metadata = client.V1ObjectMeta(
-        generate_name=job_name, labels={"kueue.x-k8s.io/queue-name": "user-queue"}
-    )
-
-    # Job container
-    container = client.V1Container(
-        image=image,
-        name="dummy-job",
-        args=args,
-        resources={
-            "requests": {
-                "cpu": 1,
-                "memory": "200Mi",
-            }
-        },
-    )
-
-    # Job template
-    template = {"spec": {"containers": [container], "restartPolicy": "Never"}}
-    return client.V1Job(
-        api_version="batch/v1",
-        kind="Job",
-        metadata=metadata,
-        spec=client.V1JobSpec(
-            parallelism=1, completions=3, suspend=True, template=template
-        ),
-    )
-
-
-def main():
-    """
-    Run a job.
-    """
-    parser = get_parser()
-    args, _ = parser.parse_known_args()
-
-    # Generate a CRD spec
-    crd = generate_job_crd(args.job_name, args.image, args.args)
-    batch_api = client.BatchV1Api()
-    print(f"📦️ Container image selected is {args.image}...")
-    print(f"⭐️ Creating sample job with prefix {args.job_name}...")
-    batch_api.create_namespaced_job("default", crd)
-    print(
-        'Use:\n"kubectl get queue" to see queue assignment\n"kubectl get jobs" to see jobs'
-    )
-
-
-if __name__ == "__main__":
-    main()
+{{% include "python/sample-job.py" %}}
```

Comment:
maybe it can be something like

```
{{% include-code "python" "python/sample-job.py %}}
```

And then the shortcode would add the ticks, and we could also put a link to the asset for download.

Reply:
Okay. I will test that out. 

It looks for files from the directory include was called. Gotta figure out if absolute path will work if they are in assets. Or maybe it can be pulled via http if its being served from assets.

That we we can also inject the url of the file.

Reply:
If it helps, I did this purely with JavaScritpt. The snippet include (for Jekyll) looks like this: https://github.com/converged-computing/flux-go/blob/main/docs/_includes/snippet.html and then the "embed" function is farily simple - it takes some set of attributes from the include (e.g., github user / repo) and then retrieves the url, adds content to the page, and uses prism to format it. https://github.com/converged-computing/flux-go/blob/30e29b410c50cefee4cdee81fac02005991d68c6/docs/_layouts/snippet-page.html#L10-L25

An example include looks like:

```
{% include snippet.html language="go" id="corego" path="pkg/core/core.go" %}
```
And that renders to:

https://converged-computing.github.io/flux-go/examples/submit/

Reply:
If we put the files in our assets folder. This should be even easier, since we know our root path. Just need /filename to get the url of the file. Keyword is "should". 😂

Reply:
Hmm, this is under https://github.com/kubernetes-sigs/kueue/tree/main/site/assets. One pet peeve I have about hugo is that everything is nested and hard to find. If this turns out to be a good design, let's make sure to make the organization explicit. E.g., perhaps have `assets/examples/python`. And then also, since it would be confusing for the user to navigate into first examples and then docs (and not find what they are looking for) it might make sense to put direct, clear links to the different examples (that would now be in different locations) so there is never any "I'm lost in the wilderness of this repository looking for that example file!" :laughing:

Reply:
And if possible, we could also link to the file directly from the example. (I do that in the flux-go example linked above, screenshot below).
![image](https://github.com/kubernetes-sigs/kueue/assets/814322/f2634c88-a511-4e2a-8672-1bed1d4a6667)